File: frontend/src/lib/providerPermissions.js
export const PROVIDER_PERMISSIONS = {
  codex: [
    ["read-only", "Read only", "Read files without changing them."],
    ["workspace-write", "Workspace write", "Edit files in the workspace and permitted folders."],
    ["danger-full-access", "Full access", "Run commands without the Codex sandbox."],
  ],
  claude_code: [
    ["default", "CLI default", "Use Claude Code's configured permissions."],
    ["manual", "Manual", "Use Claude Code's manual permission checks."],
    ["acceptEdits", "Accept edits", "Automatically approve file edits."],
    ["auto", "Auto", "Let Claude Code review tool permissions automatically."],
    ["dontAsk", "Don't ask", "Run allowed tools and deny tools that need approval."],
    ["plan", "Plan", "Explore and plan before making changes."],
    ["bypassPermissions", "Bypass permissions", "Skip Claude Code permission checks."],
  ],
};

export function defaultPermissionMode(provider) {
  return provider === "claude_code" ? "dontAsk" : "workspace-write";
}

File: src/gofer/core/provider_permissions.py
"""Permission flags shared by provider adapters and Rem."""

CODEX_PERMISSION_MODES = ("read-only", "workspace-write", "danger-full-access")
CLAUDE_PERMISSION_MODES = (
    "default",
    "acceptEdits",
    "auto",
    "manual",
    "dontAsk",
    "plan",
    "bypassPermissions",
)


def provider_permission_args(provider: str, mode: str | None) -> list[str]:
    if provider == "codex":
        mode = "workspace-write" if mode in (None, "default") else mode
        if mode not in CODEX_PERMISSION_MODES:
            raise ValueError(f"Unknown Codex permission mode '{mode}'")
        return ["--sandbox", mode]
    if provider == "claude_code":
        if mode is None or mode == "default":
            return []
        if mode not in CLAUDE_PERMISSION_MODES:
            raise ValueError(f"Unknown Claude Code permission mode '{mode}'")
        return ["--permission-mode", mode]
    raise ValueError(f"Unknown provider '{provider}'")
